util/app/path/path.go

```diff
@@ -108,8 +108,15 @@ func GetAppRefreshPaths(app *v1alpha1.Application) []string {
 			if filepath.IsAbs(item) {
 				paths = append(paths, item[1:])
 			} else {
-				for _, source := range app.Spec.GetSources() {
-					paths = append(paths, filepath.Clean(filepath.Join(source.Path, item)))
+				// For sourceHydrator apps, resolve paths relative to DRY source
```

Comment:
This method is also called in the GetRepoObjs function and should return the right value depending if we are current acting on the dry or sync source.